### PR TITLE
Add [...] menu to search results list

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add [...] menu to search results list

#### Ticket Link
Github: #4750 
Jira ticket: [PLT-4562](https://mattermost.atlassian.net/browse/PLT-4562)

#### Checklist
- [x] Has UI changes

### CODE REVIEW

File: webapp/components/search_results_item.jsx [hunk 1]
```diff
@@ -33,17 +34,36 @@ export default class SearchResultsItem extends React.Component {
         super(props);
 
         this.handleFocusRHSClick = this.handleFocusRHSClick.bind(this);
+        this.handleJumplick = this.handleJumplick.bind(this);
```
Comment (hmhealey) [inline webapp/components/search_results_item.jsx]: Small typo here. Both of these should be `handleJumpClick`

**Comment by hmhealey** [comment]
Also, I love that we have the `DotMenu` component here. That's so much nicer than having to copy and paste the whole thing from another post component

**Comment by jwilander** [comment]
@saturninoabril I think you need to rebase to match with some enterprise changes

**Comment by saturninoabril** [comment]
I temporarily hold this and set as WIP. The issue I'm seeing is pinned post does not update dynamically to RHS. When post is pinned at center, the pinned post does not update at RHS. Will do this for 4.1

**Comment by saturninoabril** [comment]
PR updated which fix updates on search results when post is  pinned or edited.

File: webapp/components/search_results_item.jsx [hunk 2]
```diff
@@ -70,14 +86,35 @@ export default class SearchResultsItem extends React.Component {
...
+                type: ActionTypes.RECEIVED_SEARCH_TERM,
+                term: null,
+                do_search: false,
+                is_mention_search: false
+            });
+
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_POST_SELECTED,
+                postId: null
+            });
```
Comment (jwilander) [inline webapp/components/search_results_item.jsx]: Can you add an action to do the app dispatches? We should avoid using AppDispatcher directly in components

Comment (saturninoabril) [inline webapp/components/search_results_item.jsx]: @jwilander noted, just used `GlobalActions.toggleSideBarAction(false);`

**Pull request merged**